### ISSUE TITLE
Queries on hyper-parameters

### ISSUE THREAD

**Issue by anuj-sharma-19** [tianweiy/CenterPoint#65]
Hi,

Thanks again for the awesome work !!

I am trying to replicate the numbers from the earlier version of `CenterPoint` paper (using `CenterPoint - PointPillars` without dcn or test-time augmentations) and am having small differences compared to the benchmark. With regards to this, I have a few queries regarding the different hyper-parameters in the code, it would be really great if you could please clarify them:

- How many GPUs should I use to train? --- The documentation and the paper says to use 4, but the model config uses 8.
- What learning rate should I use (0.001) or something else?
- Should the learning rate also be scaled using `autoscale-lr` parameter as given in train script? Also, is 0.001 learning rate to be used with 4 or 8 GPUs?
- What are the hyper-parameters for classification and regression losses? --- I can see them to be 1.0 and 0.25, but not sure if these losses are scaled further.
- I am using pytorch-lightning for multi-gpu training setup which uses `DistributedSampler`, whereas I can see the code uses `DistributedGroupSampler` for distributed training. I find them to be quite similar. Do you think this could have a large impact?
- The paper says to use `AdamW` optimizer, whereas the code uses `Adam`, so not sure which one to use?
- The paper also says to use translation noise of 0.2 m, but that does not seem to be used in the preprocessing pipeline.

I am sorry for so many queries above, but I would highly appreciate if you could kindly clarify these. Also, if you could share tensorboard logs for the training, it would be really great just to understand and compare the loss characteristics etc.

Looking forward to your reply. 

Thanks again !!

Best Regards,
Anuj

**Comment by tianweiy** [comment]
> How many GPUs should I use to train? --- The documentation and the paper says to use 4, but the model config uses 8.

It is still 4. Which config do you refer to? 

> What learning rate should I use (0.001) or something else?

0.001 for nuScenes with 4 GPUS

> Should the learning rate also be scaled using autoscale-lr parameter as given in train script? Also, is 0.001 learning rate to be used with 4 or 8 GPUs?

I didn't use the autoscale-lr parameter. 4 GPU 

> What are the hyper-parameters for classification and regression losses? --- I can see them to be 1.0 and 0.25, but not sure if these losses are scaled further.

Yes, 1 and 0.25 for classification and regression

> I am using pytorch-lightning for multi-gpu training setup which uses DistributedSampler, whereas I can see the code uses DistributedGroupSampler for distributed training. I find them to be quite similar. Do you think this could have a large impact?

Maybe, I am not sure about this.

> The paper says to use AdamW optimizer, whereas the code uses Adam, so not sure which one to use?

The current one is adamW with cyclic learning rate. You can see details in [LINK](https://github.com/tianweiy/CenterPoint/blob/master/det3d/solver/fastai_optim.py).

 > The paper also says to use a translation noise of 0.2 m, but that does not seem to be used in the preprocessing pipeline.

I found a bug so that even the old models don't use this translation augmentation. I removed it in the current version. 

> if you could share tensorboard logs for the training, it would be really great just to understand and compare the loss characteristics etc.

We only log in text files in the current codebase. I will send you the log later today.

**Comment by anuj-sharma-19** [comment]
Awesome, thank you very much for the quick response and all the clarifications. :+1: 

I was referring to `nusc_centerpoint_pp_02voxel_circle_nms.py` config in my queries above. I will have a relook into the training pipeline that I am using :)

Looking forward to your training logs.

Thanks again !!

Best,
Anuj

**Comment by tianweiy** [comment]
I only found the log for the current pp model. The old one should be really similar. The only difference in current version is that the model now uses two instead of one linear layer in pillar feature network.  
[log.zip](https://github.com/tianweiy/CenterPoint/files/5805634/log.zip)

**Comment by anuj-sharma-19** [comment]
Thanks a lot for the logs. It is super helpful, will use the losses to trace the issue. :+1:

**Comment by tianweiy** [comment]
close due to inactivity
